GitHubRepositories.ps1

```diff
@@ -2944,6 +2945,593 @@ filter Set-GitHubRepositoryActionsPermission
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamSlugUri')]
+        [Alias('RepositoryUrl')]
+        [string] $Uri,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamNameElements')]
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamSlugElements')]
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamNameUri')]
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamSlugUri')]
+        [ValidateNotNullOrEmpty()]
+        [string] $OrganizationName,
+
+        [Parameter(
+            Mandatory,
+            ParameterSetName = 'TeamNameElements')]
+        [Parameter(
+            Mandatory,
+            ParameterSetName = 'TeamNameUri')]
+        [ValidateNotNullOrEmpty()]
+        [string] $TeamName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamSlugElements')]
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamSlugUri')]
+        [ValidateNotNullOrEmpty()]
+        [string] $TeamSlug,
+
+        [string] $AccessToken
+    )
+
+    Write-InvocationLog
+
+    $telemetryProperties = @{}
+
+    $elements = Resolve-RepositoryElements
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    if ($PSBoundParameters.ContainsKey('TeamName'))
+    {
+        $team = Get-GitHubTeam -OrganizationName $OrganizationName |
+            Where-Object -Property name -eq $TeamName
+
+        if ($null -eq $team)
+        {
+            $message = "Team '$TeamName' not found"
+            Write-Log -Message $message -Level Error
+            throw $message
+        }
+        else
+        {
+            $TeamSlug = $team.slug
+        }
+    }
+
+    $telemetryProperties['TeamSlug'] = Get-PiiSafeString -PlainText $TeamSlug
+
+    $uriFragment = "/orgs/$OrganizationName/teams/$TeamSlug/repos/$OwnerName/$RepositoryName"
+    $description = "Getting team $TeamSlug permissions for repository $RepositoryName"
+
+    $params = @{
+        UriFragment = $uriFragment
+        Description =  $description
+        AcceptHeader = $script:repositoryAcceptHeader
+        Method = 'Get'
+        AccessToken = $AccessToken
+        TelemetryEventName = $MyInvocation.MyCommand.Name
+        TelemetryProperties = $telemetryProperties
+    }
+
+    $result = Invoke-GHRestMethod @params
+
+    if ($PSBoundParameters.ContainsKey('TeamSlug'))
+    {
+        $team = Get-GitHubTeam -OrganizationName $OrganizationName -TeamSlug $TeamSlug
+
+        $TeamName = $team.name
+    }
+
+    return (Invoke-GHRestMethod @params |
```

Comment:
It looks like you're executing the command twice (once here, and then once previously at 3125) unless I'm missing something.  Shouldn't you just be reusing `$result`?

```suggestion
    return ($result |
```